frontend/src/components/camp/picasso/Picasso.vue

```diff
@@ -0,0 +1,414 @@
+      timeMouseMove,
+      timeMouseUp,
+      nativeMouseUp,
+      extendBottom,
+
+      // data
+      isSaving,
+      patchError
+    }
+  },
+  data () {
+    return {
+      maxDays: 100,
+      entryWidth: 80,
+      value: '',
+      activitiesLoading: true
+    }
+  },
+  computed: {
+    widthPluralization () {
+      if (this.entryWidth < 81) {
+        return 0
+      } else if (this.entryWidth < 85) {
+        return 1
+      } else {
+        return 2
+      }
+    },
+    now () {
+      return this.$refs.calendar ? this.$refs.calendar.times.now : null
+    },
+    dateNow () {
+      return this.$refs.calendar ? this.$refs.calendar.times.now.date : null
+    },
+    nowY () {
+      return this.$refs.calendar ? this.$refs.calendar.timeToY(this.now) + 'px' : '-10px'
+    },
+    camp () {
+      return this.period.camp()
+    },
+    computedIntervalHeight () {
+      return this.intervalHeight !== 0 ? this.intervalHeight : this.$vuetify.breakpoint.xsOnly ? (this.$vuetify.breakpoint.height - 140) / 19 : Math.max((this.$vuetify.breakpoint.height - 204) / 19, 32)
+    }
+  },
+  mounted () {
+    this.api.get().activities({ period: this.period._meta.self })._meta.load.then(() => { this.activitiesLoading = false })
+  },
+  methods: {
+    resize () {
+      const widthIntervals = 46
+      this.entryWidth = Math.max((this.$refs.calendar.$el.offsetWidth - widthIntervals) / this.$refs.calendar.days.length, 80)
+    },
+    getActivityName (scheduleEntry, _) {
+      if (this.isActivityLoading(scheduleEntry)) return this.$tc('global.loading')
+      return (scheduleEntry.number ? scheduleEntry.number + ' ' : '') +
+        (scheduleEntry.activity().category().short ? scheduleEntry.activity().category().short + ': ' : '') +
+        scheduleEntry.activity().title
+    },
+    getActivityColor (scheduleEntry, _) {
+      if (this.isCategoryLoading(scheduleEntry)) return 'grey lighten-1'
+      const color = scheduleEntry.activity().category().color
+      return isCssColor(color) ? color : color + ' elevation-4 v-event--temporary'
+    },
+    isActivityLoading (scheduleEntry) {
+      return this.activitiesLoading || (scheduleEntry.activity()?._meta?.loading ?? false)
+    },
+    isCategoryLoading (scheduleEntry) {
+      return scheduleEntry.activity().category()?._meta?.loading ?? false
+    },
+    intervalFormat (time) {
+      return this.$date.utc(time.date + ' ' + time.time).format(this.$tc('global.datetime.hourLong'))
+    },
+    dayFormat (day) {
+      if (this.$vuetify.breakpoint.smAndDown) {
+        return this.$date.utc(day.date).format(this.$tc('global.datetime.dateShort'))
+      } else {
+        return this.$date.utc(day.date).format(this.$tc('global.datetime.dateLong'))
+      }
+    },
+    weekdayFormat () {
+      return ''
+    },
+
+    entryClick ({ event: entry }) {
+      // if in read-only mode --> fire openEntry event
+      // in edit mode, the event is already fired by drag & drop logic
```

Comment:
Bin nicht ganz überzeugt dass das DragAndDrop Composable jemals für den `openEntry` Event verantwortlich sein sollte. Klar, für beides muss man auf Mausklicks hören, aber inhaltlich ist das etwas ganz separates.

Reply:
Ja, das habe ich mir auch schon so gedacht :-)

Ich versuch mal einen Wurf das zu entflechten.

Reply:
Implemented with https://github.com/ecamp/ecamp3/pull/2334/commits/cee43fcadc5c5749c6f1c9b0b5ebce9480e8fb13

Thereotisch könne man noch einen Schritt weiter gehen und das drag & drop in 3 separate composables splitten:
- drag & drop (move event)
- resize
- create new